src/pubkey.h

```diff
@@ -187,6 +188,81 @@ class CPubKey
     bool Derive(CPubKey& pubkeyChild, ChainCode &ccChild, unsigned int nChild, const ChainCode& cc) const;
 };
 
+class CCompressedPubKey {
+private:
+    CPubKey pubKey;
+public:
+
+    CCompressedPubKey()
+    {
+        // pubKey's 0-argument constructor invalidates it.
+    }
+
+    CCompressedPubKey(const CPubKey &pubKey)
+    {
+        this->pubKey = pubKey;
+        // TODO: check that it's compressed and valid and throw exception if
+        // not.
+    }
+
+    unsigned int GetSerializeSize(int nType, int nVersion) const
+    {
+        assert(pubKey.size() == 33);
+        return pubKey.size();
```

Comment:
This code is removed by a future commit in this PR. (We don't need a `CCompressedPubKey` class anymore since we're using ed25519.)